getValidation/index.js

```diff
@@ -1,173 +1,214 @@
+    }), // Note updates verified status to true or false, null status indicates never attempted verification
+  });
+  var resultJSON = await result.json();
+
+  var thisVerificationAttempt = verificationAttempt;
+  thisVerificationAttempt.userUpdated = resultJSON.success;
+  thisVerificationAttempt.verified = thisVerificationAttempt.verified && thisVerificationAttempt.userUpdated
+  return thisVerificationAttempt;
+};
+
+const returnResponse = function (context, response, responseError) {
+  // Generate HTTP response
+  var statusCode;
+  if (responseError) {
+    statusCode = 500;
+  } else {
+    statusCode = response.verified ? 200 : 401;
+  }
+
+  var responseBody;
+  if (responseError) {
+    responseBody = respondWithError ? responseError : null;
+  } else {
+    responseBody = response;
+  }
+
+  context.res = {
+    status: statusCode,
+    body: responseBody,
+    headers: {
+      "Content-Type": "application/json",
+    },
+  };
+};
+
+const getYotiDetails = function (activityDetails) {
+  if (activityDetails) {
+    const rememberMeId = activityDetails.getRememberMeId();
+    const profile = activityDetails.getProfile();
+    const dob = profile.getDateOfBirth().getValue();
+    const fullName = profile.getFullName().getValue();
+    return { name: fullName, dob: dob, rememberMeId: rememberMeId };
+  }
+};
+
+const getYoti = function (token) {
+  return yotiClient.getActivityDetails(token);
+};
+
+const getUser = async function (userId) {
+  var response = await fetch(getUserServiceUrl + "?ID=" + userId, {
+    method: "get",
+    headers: {
+      "content-type": "application/json",
+      "x-functions-key": userServiceKey,
+    },
+  });
+
+  return response.json();
+};
+
+const processDetails = function (yoti, user) {
+  const yotiResponse = getYotiDetails(yoti);
+  const userResponse = user.user;
+  return { yoti: yotiResponse, user: userResponse };
+};
+
+const completeVerification = function (context, response, responseError) {
+  // Save to Cosmos
+  if (responseError) {
+    context.bindings.outputDocument = {
+      datestamp: new Date().toISOString(),
+      isError: true,
+      error: errorCosmos ? responseError : "An error occurred, but security settings prevented database logging",
+    };
+  } else {
+    context.bindings.outputDocument = response;
+  }
+
+  // Respond to client
+  returnResponse(context, response, responseError);
+
+  // Context Finishes
+  context.done();
+};
+
+module.exports = function (context, req) {
+  const yotiPromise = getYoti(req.params.token);
+  const userPromise = getUser(req.params.userId);
+  var logOutput = "";
+  Promise.all([yotiPromise, userPromise])
+    .then((values) => {
+      logOutput += "Post-Values ";
+      return processDetails(values[0], values[1]);
+    })
+    .then((details) => {
+      logOutput += "Post-Details ";
+      return getVerificationData(details.yoti, details.user);
+    })
+    .then((data) => {
+      logOutput += "Post-verification ";

```

Comment:
If these logs are going to be useful, consider calling context.log each time.
Benefits:
* We'll get a timestamp for each one so we can investigate any delays
* We'll get the logs even if code execution stops for any reason
(This would mean we get logs in success and failure cases, which is not how you've written it.)

Reply:
Looks like we might now be logging the same stuff over and over...
Post-Values 
Post-Values Post-Details 
Post-Values Post-Details Post-verification 
...